first commit of a CIP for changes to on-chain governance votes (#35)
global-synchronizer-foundation/cips: cip-0051/cip-0051.md

## CIP 0051

<pre>
  CIP: CIP 0051
  Title: Streamline onchain governance votes
  Author: Itai Segall
  Comments-Summary: No comments yet
  Status: Draft
  Type: Governance
  Created: 2025-03-04
  License: CC0-1.0
</pre>

## Abstract

This CIP proposes the following changes to on-chain voting:

a) votes on config changes will change individual (or a few) config values, and not the whole set of configurations;

b) vote proposals will have an expiration date, after which if not enough votes have been submitted,
they will be auto-rejected;

> c) votes will remain open for editing until they become effective;

Reviewer: change to "votes on a proposal may be changed until the vote proposal becomes effective"

d) votes will be early-rejected upon 2/3 majority of reject votes;

e) vote requests can be labeled as "immediate effectivity",
meaning they will automatically get accepted and take effect immediately as soon as 2/3 of the
Super Validator nodes vote in favor.

## Specification

### Vote on Individual Configuration Values

In the new implementation, a vote request will only apply changes to those fields that the
proposer has chosen to change, leaving the others unmodified, regardless of their value at the
time of applying the change.

### Expiration as a Due Date for Enough Votes

Every vote request will have an expiration date. If at the time of expiration, not enough
votes are in (i.e. less than 2/3 of the Super Validators have put in any vote), then the
vote request is immediately considered stale and auto-rejected.

> ### Votes Remain Open

Reviewer: Change to "Vote Proposals Remain Open"

> The new implementation will allow votes to be modified until the point where the action being

Reviewer: "The new implementation will allow votes on a proposal to be modified until the point where the action being"

> vote is scheduled to go into effect.

Reviewer: "voted on"

> 

Reviewer: Add:

The text of a proposal is locked once it is submitted and cannot be modified during voting. The proposal can only be accepted or rejected, or it may expire.

## CIP 0051

<pre>
  CIP: CIP 0051
  Title: Streamline onchain governance votes
  Author: Itai Segall
  Comments-Summary: No comments yet
  Status: Draft
  Type: Governance
  Created: 2025-03-04
  License: CC0-1.0
</pre>

## Abstract

This CIP proposes the following changes to on-chain voting:

a) votes on config changes will change individual (or a few) config values, and not the whole set of configurations;

b) vote proposals will have an expiration date, after which if not enough votes have been submitted,
they will be auto-rejected;

c) votes on a proposal may be changed until the vote proposal becomes effective;

d) votes will be early-rejected upon 2/3 majority of reject votes;

e) vote requests can be labeled as "immediate effectivity",
meaning they will automatically get accepted and take effect immediately as soon as 2/3 of the
Super Validator nodes vote in favor.

## Specification

> ### Vote on Individual Configuration Values

Reviewer 2: ```suggestion
### Make it be like this instead
```

In the new implementation, a vote request will only apply changes to those fields that the
proposer has chosen to change, leaving the others unmodified, regardless of their value at the
time of applying the change.

### Expiration as a Due Date for Enough Votes

Every vote request will have an expiration date. If at the time of expiration, not enough
votes are in (i.e. less than 2/3 of the Super Validators have put in any vote), then the
vote request is immediately considered stale and auto-rejected.

### Vote Proposals Remain Open

The new implementation will allow votes on a proposal to be modified until the point where the action being
voted on is scheduled to go into effect.

The text of a proposal is locked once it is submitted and cannot be modified during voting. The proposal can only be accepted or rejected, or it may expire.

### Early Rejection

In the new implementation, vote requests that collected 2/3 "Reject" votes will immediately
close as rejected.

### Immediate Effectivity

> The new implementation will introduce an "immediate effectivity" option for vote requests,
> which does not specify an explicit effectivity date, but instead becomes effective immediately
> upon collecting 2/3 "Accept" votes.

Reviewer: ```suggestion
The new implementation will introduce an "immediate effectivity" option for vote proposals,
which does not specify an explicit effectivity date, but instead becomes effective immediately
upon collecting 2/3 "Accept" votes.

Vote proposals that use immediate effectivity would be independent of the expiration date, and may happen prior to the expiration date.  

If there is a desire for the vote proposal to either become effective or expire at the same date-time as its expiration, the submitter may set the "effective date" to the same date-time as the "expiration date"
```

> upon collecting 2/3 "Accept" votes, but not earlier than the expiration date.

Reviewer: ```suggestion
upon collecting 2/3 "Accept" votes, but not earlier than the expiration date.
```

> upon collecting 2/3 "Accept" votes, but not earlier than the expiration date.

Reviewer: ```suggestion
upon collecting 2/3 "Accept" votes.
```
